src/plugins/intel_npu/src/plugin/npuw/compiled_model.cpp

```diff
@@ -147,6 +147,14 @@ ov::npuw::CompiledModel::CompiledModel(const std::shared_ptr<ov::Model>& model,
         rewr.run_on_model(model);
     }
 
+    // Add Slice before last MatMul for the prefill model
+    ov::pass::GraphRewrite rewr;
+    rewr.add_matcher<ov::npuw::patterns::opt::SliceLastMatmul>();
```

Comment:
Let's make this under a config too. We're breaking the model's API convention in fact, user gave us a model with `[N,K]` output, but the compiled model he gets instead would be `[1,K]`.

Let it be `NPUW_SLICE_OUT`, let it be NO by default, let @TolyaTalamanov and our internal configs pass it by default.

Reply:
Done